### PR TITLE
Changing dacFxService and schemaCompareService to use shared logic for error handling

### PR DESCRIPTION
**Pull request by Benjin**
Expanding the helper method that's already being used for `sqlProjectService` to `dacFxService` and `schemaCompareService`, just to test the waters a little more with extensions I'm more familiar with.  Once this is in, I'll convert the remaining services.

### CODE REVIEW

**Review by kisantia — APPROVED** [review]

File: extensions/mssql/src/baseService.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+export abstract class BaseService {
+	constructor(protected readonly client: SqlOpsDataClient) { }
+
+	/**
+	 * Runs the specified request wrapped in the requisite try-catch
+	 * @param type RequestType, typically in the format 'contracts.DoThingRequest.type'
+	 * @param params parameters to be passed to the request
+	 * @returns result from the request
+	 */
+	protected async runWithErrorHandling<P, R, E, RO>(type: RequestType<P, R, E, RO>, params: P): Promise<R> {
```
Comment (alanrenmsft) [inline extensions/mssql/src/baseService.ts:21]: I am sure others would appreciate if you could update all the services :)

**Review by alanrenmsft — APPROVED** [review]

File: extensions/mssql/src/baseService.ts (hunk 1 above)
Comment (Benjin) [reply extensions/mssql/src/baseService.ts:21]: That's the plan!  Just wanted to start with the services I knew best in case y'all spotted any issues.

Comment (alanrenmsft) [reply extensions/mssql/src/baseService.ts:21]: I think we can probably define in the dataprotocol-client so that even other extensions can use this.

Comment (Charles-Gagnon) [reply extensions/mssql/src/baseService.ts:21]: +1 on that, would love to see that added there

**Review by Charles-Gagnon — APPROVED** [review]

File: extensions/mssql/src/baseService.ts (hunk 1 above)
Comment (Benjin) [reply extensions/mssql/src/baseService.ts:21]: Good call!  I'll add it there, and swap this over once the new version goes out.

**Pull request merged**